### PR TITLE
Appsembler Apps - Analytics - Juniper Upgrade

### PR DESCRIPTION
**Pull request by johnbaldwin**
- Enforced boolean typ on the 'SHOW_X' (where X is the APP label) boolean flags
- Added test coverage
- Reorganized tests under 'tests' folder in 'appsembler.analytics'

https://appsembler.atlassian.net/browse/RED-1601

### CODE REVIEW

File: openedx/core/djangoapps/appsembler/analytics/context_processors.py [hunk 1]
```diff
@@ -4,13 +4,18 @@
...
 
 def google_analytics(request):
+    app_id = getattr(settings, 'GOOGLE_ANALYTICS_APP_ID', None)
+    show_app = True if app_id else False
+
     data = {
-        'GOOGLE_ANALYTICS_APP_ID': getattr(settings, 'GOOGLE_ANALYTICS_APP_ID', None),
-        'SHOW_GOOGLE_ANALYTICS': getattr(settings, 'GOOGLE_ANALYTICS_APP_ID', None),
+        'GOOGLE_ANALYTICS_APP_ID': app_id,
+        'SHOW_GOOGLE_ANALYTICS': show_app,
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/analytics/context_processors.py:18]: `show_app = True if app_id else False` doesn't seem to be Pythonic. `bool()` should do the same job.

```suggestion
        'SHOW_GOOGLE_ANALYTICS': bool(show_app),
```

File: openedx/core/djangoapps/appsembler/analytics/context_processors.py [hunk 2]
```diff
@@ -4,13 +4,18 @@
...
 
-from openedx.core.djangoapps.appsembler.analytics.helpers import should_show_hubspot
+from openedx.core.djangoapps.appsembler.analytics.helpers import (
+    should_show_hubspot
+)
 
 
 def google_analytics(request):
+    app_id = getattr(settings, 'GOOGLE_ANALYTICS_APP_ID', None)
+    show_app = True if app_id else False
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/analytics/context_processors.py:14]:
```suggestion
```

File: openedx/core/djangoapps/appsembler/analytics/context_processors.py [hunk 3]
```diff
@@ -23,17 +28,23 @@ def google_analytics(request):
 
 
 def mixpanel(request):
+    app_id = getattr(settings, 'MIXPANEL_APP_ID', None)
+    show_app = True if app_id else False
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/analytics/context_processors.py:32]:
```suggestion
```

File: openedx/core/djangoapps/appsembler/analytics/context_processors.py [hunk 4]
```diff
@@ -23,17 +28,23 @@ def google_analytics(request):
...
     data = {
-        'MIXPANEL_APP_ID': getattr(settings, 'MIXPANEL_APP_ID', None),
-        'SHOW_MIXPANEL': getattr(settings, 'MIXPANEL_APP_ID', None),
+        'MIXPANEL_APP_ID': app_id,
+        'SHOW_MIXPANEL': show_app,
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/analytics/context_processors.py:35]: Ditto

```suggestion
        'SHOW_MIXPANEL': bool(app_id),
```

File: openedx/core/djangoapps/appsembler/analytics/context_processors.py [hunk 5]
```diff
@@ -23,17 +28,23 @@ def google_analytics(request):
...
     }
     return data
 
 
 def hubspot(request):
     hubspot_portal_id = getattr(settings, 'HUBSPOT_PORTAL_ID', None)
+    if hubspot_portal_id and should_show_hubspot(request.user):
+        show_app = True
+    else:
+        show_app = False
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/analytics/context_processors.py:45]:
```suggestion
    show_app = bool(hubspot_portal_id) and should_show_hubspot(request.user)
```

**Review by OmarIthawi — APPROVED** [review]
Thanks @johnbaldwin. LGTM. 

I've suggested to avoid the `True if app_id else False` in favor of `bool()` because it's more succinct and takes only a second to be understood.

**Comment by johnbaldwin** [comment]
@OmarIthawi If you would please give the context processers another look over if you like the update

**Review by OmarIthawi — APPROVED** [review]

**Review by OmarIthawi — APPROVED** [review]
thanks @johnbaldwin for addressing the style notes. looks great!

**Pull request merged**